From 8554e892fa64d8e1f82abcb5c2f5981040d54250 Mon Sep 17 00:00:00 2001
From: Shizhao Liu <lshizhao@amazon.com>
Date: Wed, 1 Oct 2025 17:54:37 -0700
Subject: [PATCH 16/16] Vsphere Clone Builder support

1. Add tasks to disable firewall service on Rhel and Ubuntu.
If base template has firewall enabled, it will block certain
ports used by k8s api-servier, kubelet, etc. Disable firewall
during build process, which aligns with what vsphere-iso build
is doing.

Signed-off-by: Shizhao Liu <lshizhao@amazon.com>
---
 images/capi/ansible/roles/setup/tasks/debian.yml | 7 +++++++
 images/capi/ansible/roles/setup/tasks/redhat.yml | 7 +++++++
 2 files changed, 14 insertions(+)

diff --git a/images/capi/ansible/roles/setup/tasks/debian.yml b/images/capi/ansible/roles/setup/tasks/debian.yml
index 41e0fd631..9205872ba 100644
--- a/images/capi/ansible/roles/setup/tasks/debian.yml
+++ b/images/capi/ansible/roles/setup/tasks/debian.yml
@@ -128,3 +128,10 @@
 - name: Remove '--no-tpm --no-efivars' from nullboot post install script
   ansible.builtin.command: sed -i 's/nullbootctl --no-tpm --no-efivars/nullbootctl/' /var/lib/dpkg/info/nullboot.postinst
   when: packer_build_name is search('cvm')
+
+- name: Disable ufw service
+  ansible.builtin.systemd:
+    name: ufw
+    state: stopped
+    enabled: false
+  when: ansible_distribution == "Ubuntu"
diff --git a/images/capi/ansible/roles/setup/tasks/redhat.yml b/images/capi/ansible/roles/setup/tasks/redhat.yml
index 2a98dae54..820f3a714 100644
--- a/images/capi/ansible/roles/setup/tasks/redhat.yml
+++ b/images/capi/ansible/roles/setup/tasks/redhat.yml
@@ -164,3 +164,10 @@
     state: present
     lock_timeout: 60
   when: ansible_os_family == "RedHat" and ansible_distribution == "RedHat" and ansible_distribution_major_version == "9"
+
+- name: Disable firewalld service
+  ansible.builtin.systemd:
+    name: firewalld
+    state: stopped
+    enabled: false
+  when: ansible_service_mgr == "systemd"
-- 
2.49.0

